### PR TITLE
Add some basic functionality to demo app

### PR DESCRIPTION
**Pull request by procchio6**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -4,18 +4,19 @@ module.exports = {
   entry: './src/index',
   output: {
     path: path.resolve(__dirname, 'dist'),
-    filename: 'bundle.js'
+    filename: 'index.js',
+    libraryTarget: 'commonjs2',
```
Comment (procchio6) [inline webpack.config.js]: Needed to change filename and have webpack create a commonjs library.

File: tsconfig.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "exclude": [
     "demo",
     "dist",
-    "node_modules"
+    "node_modules",
+    "test"
```
Comment (procchio6) [inline tsconfig.json]: Don't include test directory when producing type declarations.

File: package.json [hunk 3]
```diff
@@ -3,12 +3,13 @@
...
   "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "types": "dist/types/index.d.ts",
   "files": [
     "dist/**/*"
   ],
   "scripts": {
-    "build": "npm run clean && webpack",
+    "build": "npm run clean && npm run build:types && webpack",
+    "build:types": "tsc --emitDeclarationOnly --outdir dist/types",
```
Comment (procchio6) [inline package.json]: Create type declaration during build.

File: webpack.config.js (hunk 1 above)
Comment (brianmcallister) [reply webpack.config.js]: Riiiiiight, this is running in a browser. Makes sense.

**Pull request merged**